QuickReference.ipynb

```diff
@@ -0,0 +1,402 @@
+    "results.to_table()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Download an image\n",
+    "For the first result, print the file format and download the file. If repeatedly executing this code, add `cache=True` to `download_file()` to prevent repeated downloads.\n",
+    "\n",
+    "See `download_file()` documentation here: https://docs.astropy.org/en/stable/api/astropy.utils.data.download_file.html#astropy.utils.data.download_file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(results[0].format)\n",
+    "file_name = download_file(results[0].getdataurl())  \n",
+    "file_name"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2.4 Spectral search\n",
+    "Example:  Find a spectral service for x-ray data.  Query it around Delta Ori with a search **diameter** of 10 arc minutes, and download the first data product.  Note that the results table can be inspected for potentially useful columns.\n",
+    "\n",
+    "Spectral search is very similar to image search. In this example, note:\n",
+    "* **`diameter`** defines the size of the search region\n",
+    "* `waveband` used in `regsearch()`\n",
+    "* Astropy `Angle` used to specify radius units other than degrees."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Search for a spectrum search service that has x-ray data.\n",
+    "services = vo.regsearch(servicetype='spectrum', waveband='x-ray')\n",
+    "\n",
+    "# Assuming there are services and the first one is OK...\n",
+    "results = services[0].search(pos=SkyCoord.from_name(\"Delta Ori\"), \n",
+    "                             diameter=Angle(10 * u.arcmin))\n",
+    "\n",
+    "# Assuming there are results, download the first file.\n",
+    "print(f'Title: {results[0].title}, Format: {results[0].format}')\n",
+    "file_name = download_file(results[0].getdataurl())  \n",
+    "file_name"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2.5 Table search\n",
+    "Example:  Find the HEASARC Table Access Protocol (TAP) service, get some information about the available tables."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "services = vo.regsearch(servicetype='tap', keywords='heasarc')\n",
+    "print(f'{len(services)} service(s) found.')\n",
+    "\n",
+    "# We found only one service.  Print some info about the service and its tables.\n",
+    "print(f'Title: {services[0].res_title}')\n",
+    "print(f'{services[0].res_description}')\n",
+    "      \n",
+    "tables = services[0].service.tables  # Queries for details of the service's tables\n",
+    "print(f'{len(tables)} tables:')\n",
+    "for t in tables:\n",
```

Comment:
This works for HEASARC which has populated all the descriptions. For IRSA TAP I think these are all None so may raise an exception.

Reply:
I checked it with IRSA, and it doesn't raise an exception.  It just shows None for the table descriptions, which seems fine.  Column descriptions seem populated for the tables I checked.

Reply:
Oh, that is terrific! It was something else in pyvo which failed on the None descriptions.